crates/turborepo-lib/src/global_deps_package_detector.rs

```diff
@@ -0,0 +1,146 @@
+use thiserror::Error;
+use turbopath::AnchoredSystemPath;
+use turborepo_repository::{
+    change_mapper::{DefaultPackageDetector, PackageDetection, PackageDetector},
+    package_graph::{PackageGraph, WorkspacePackage},
+};
+use wax::{BuildError, Program};
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    InvalidFilter(#[from] BuildError),
+}
+
+/// A package detector that uses a global deps list to determine
+/// if a file should cause all packages to be marked as changed.
+/// This is less conservative than the `DefaultPackageDetector`
+/// which assumes that any changed file that is not in a package
+/// is a root dependency. Since we have a list of global deps,
```

Comment:
I'm having trouble parsing this comment is "which" being used to explain `DefaultPackageDetector` or `GlobalDepsPackageDetector`?
```suggestion
/// This is less conservative than the `DefaultPackageDetector`
/// which assumes that any changed file that is not in a package
/// changes all packages. Since we have a list of global deps,
```